File: templates/_composition/_shared/Feature.DragAndDropVB/Services/DragAndDrop/VisualDropConfiguration.vb
Namespace Services.DragAndDrop

    Public Class VisualDropConfiguration
        Inherits DependencyObject

        Public Shared ReadOnly CaptionProperty As DependencyProperty = DependencyProperty.Register("Caption", GetType(String), GetType(VisualDropConfiguration), New PropertyMetadata(String.Empty))

        Public Shared ReadOnly IsCaptionVisibleProperty As DependencyProperty = DependencyProperty.Register("IsCaptionVisible", GetType(Boolean), GetType(VisualDropConfiguration), New PropertyMetadata(True))

        Public Shared ReadOnly IsContentVisibleProperty As DependencyProperty = DependencyProperty.Register("IsContentVisible", GetType(Boolean), GetType(VisualDropConfiguration), New PropertyMetadata(True))

        Public Shared ReadOnly IsGlyphVisibleProperty As DependencyProperty = DependencyProperty.Register("IsGlyphVisible", GetType(Boolean), GetType(VisualDropConfiguration), New PropertyMetadata(True))

        Public Shared ReadOnly DragStartingImageProperty As DependencyProperty = DependencyProperty.Register("DragStartingImage", GetType(ImageSource), GetType(VisualDropConfiguration), New PropertyMetadata(Nothing))

        Public Shared ReadOnly DropOverImageProperty As DependencyProperty = DependencyProperty.Register("DropOverImage", GetType(ImageSource), GetType(VisualDropConfiguration), New PropertyMetadata(Nothing))

        Public Property Caption As String
            Get
                Return CStr(GetValue(CaptionProperty))
            End Get

            Set(value As String)
                SetValue(CaptionProperty, value)
            End Set
        End Property

        Public Property IsCaptionVisible As Boolean
            Get
                Return CBool(GetValue(IsCaptionVisibleProperty))
            End Get

            Set(value As Boolean)
                SetValue(IsCaptionVisibleProperty, value)
            End Set
        End Property

        Public Property IsContentVisible As Boolean
            Get
                Return CBool(GetValue(IsContentVisibleProperty))
            End Get

            Set(value As Boolean)
                SetValue(IsContentVisibleProperty, value)
            End Set
        End Property

        Public Property IsGlyphVisible As Boolean
            Get
                Return CBool(GetValue(IsGlyphVisibleProperty))
            End Get

            Set(value As Boolean)
                SetValue(IsGlyphVisibleProperty, value)
            End Set
        End Property

        Public Property DragStartingImage As ImageSource
            Get
                Return CType(GetValue(DragStartingImageProperty), ImageSource)
            End Get

            Set(value As ImageSource)
                SetValue(DragStartingImageProperty, value)
            End Set
        End Property

        Public Property DropOverImage As ImageSource
            Get
                Return CType(GetValue(DropOverImageProperty), ImageSource)
            End Get

            Set(value As ImageSource)
                SetValue(DropOverImageProperty, value)
            End Set
        End Property
    End Class
End Namespace

File: code/TemplateStudioForUWP/Templates/_comp/_shr/DragAndDrop/Services/DragAndDrop/VisualDropConfiguration.cs
using Windows.UI.Xaml;
using Windows.UI.Xaml.Media;

namespace Param_RootNamespace.Services.DragAndDrop
{
    public class VisualDropConfiguration : DependencyObject
    {
        public static readonly DependencyProperty CaptionProperty =
            DependencyProperty.Register("Caption", typeof(string), typeof(VisualDropConfiguration), new PropertyMetadata(string.Empty));

        public static readonly DependencyProperty IsCaptionVisibleProperty =
            DependencyProperty.Register("IsCaptionVisible", typeof(bool), typeof(VisualDropConfiguration), new PropertyMetadata(true));

        public static readonly DependencyProperty IsContentVisibleProperty =
            DependencyProperty.Register("IsContentVisible", typeof(bool), typeof(VisualDropConfiguration), new PropertyMetadata(true));

        public static readonly DependencyProperty IsGlyphVisibleProperty =
            DependencyProperty.Register("IsGlyphVisible", typeof(bool), typeof(VisualDropConfiguration), new PropertyMetadata(true));

        public static readonly DependencyProperty DragStartingImageProperty =
            DependencyProperty.Register("DragStartingImage", typeof(ImageSource), typeof(VisualDropConfiguration), new PropertyMetadata(null));

        public static readonly DependencyProperty DropOverImageProperty =
            DependencyProperty.Register("DropOverImage", typeof(ImageSource), typeof(VisualDropConfiguration), new PropertyMetadata(null));

        public string Caption
        {
            get { return (string)GetValue(CaptionProperty); }
            set { SetValue(CaptionProperty, value); }
        }

        public bool IsCaptionVisible
        {
            get { return (bool)GetValue(IsCaptionVisibleProperty); }
            set { SetValue(IsCaptionVisibleProperty, value); }
        }

        public bool IsContentVisible
        {
            get { return (bool)GetValue(IsContentVisibleProperty); }
            set { SetValue(IsContentVisibleProperty, value); }
        }

        public bool IsGlyphVisible
        {
            get { return (bool)GetValue(IsGlyphVisibleProperty); }
            set { SetValue(IsGlyphVisibleProperty, value); }
        }

        public ImageSource DragStartingImage
        {
            get { return (ImageSource)GetValue(DragStartingImageProperty); }
            set { SetValue(DragStartingImageProperty, value); }
        }

        public ImageSource DropOverImage
        {
            get { return (ImageSource)GetValue(DropOverImageProperty); }
            set { SetValue(DropOverImageProperty, value); }
        }
    }
}
